analyzer-core/src/main/java/ffb/analyzer/core/queue/Task.java

```diff
@@ -0,0 +1,43 @@
+package ffb.analyzer.core.queue;
+
+import java.util.function.Consumer;
+
+public class Task<T> {
```

Comment:
What are your thoughts on binding an `Event` to a task? (We can call it something else to avoid confusion with Java events).

Instead of passing a consumer, we pass in the event that the task should perform, and the call `event#run` or something. This has the added benefit of delegating the priority of a task to a given event, since it's not really the task that's prioritized, it's what the task needs to do.